Realm/RLMObservation.hpp

```diff
@@ -114,6 +114,9 @@ class RLMObservationInfo {
 
     template<typename F>
     void forEach(F&& f) const {
+        // Ensure that `this` continues to exist until the end of the function
+        // by retaining its owner
+        __attribute__((objc_precise_lifetime)) id self = object;
```

Comment:
We could use Foundation's `NS_VALID_UNTIL_END_OF_SCOPE` instead of using the attribute directly.